applications/neuroscan/backend/api/contact/services/contact.js

```diff
@@ -75,17 +65,25 @@ module.exports = {
     let query = getBaseSearchQuery(timepoint);
 
     if (terms.length > 0) {
-      query.orderByRaw("CASE WHEN LOWER(neuronA_content.uid) LIKE ? THEN 0 ELSE 1 END, neuronA_content.uid",
-        [`%${terms[0].toLowerCase()}%`]);
+      query.orderByRaw(`
+      CASE
+        WHEN LOWER(neuronA.uid) LIKE ? THEN 0
+        ELSE 1
+      END,
+      neuronA.uid ASC,
+      neuronB.uid ASC
+    `, [`%${terms[0].toLowerCase()}%`]);
 
       query = applySearchConditions(query, terms);
     }else {
-      query.orderBy('neuronA_content.uid', 'asc');
+      query.orderBy('neuronA.uid', 'asc');
+      query.orderBy('neuronB.uid', 'asc');
     }
 
     query.offset(start).limit(limit);
 
-    return query;
+    const ids = await query;
+    return strapi.query('contact').find({ id_in: ids.map(x => x.id), _sort: 'uid' });
```

Comment:
Isn't the sort by 'uid' breaking the custom logic to sort the contacts?

Reply:
@afonsobspinto nice catch, thanks/ I will change that

Reply:
done